pkg/reconciler/cloudeventingress/cloudeventingress.go

```diff
@@ -0,0 +1,165 @@
+)
+
+// Add creates a new CloudEventIngress Controller and adds it to the Manager with
+// default RBAC. The Manager will set fields on the Controller and Start it when
+// the Manager is Started.
+func Add(mgr manager.Manager) error {
+	raImage, defined := os.LookupEnv(raImageEnvVar)
+	if !defined {
+		return fmt.Errorf("required environment variable %q not defined", raImageEnvVar)
+	}
+
+	log.Println("Adding the CloudEventIngress controller.")
+	p := &sdk.Provider{
+		AgentName: controllerAgentName,
+		Parent:    &v1alpha1.CloudEventIngress{},
+		Owns:      []runtime.Object{&servingv1alpha1.Service{}},
+		Reconciler: &reconciler{
+			scheme:              mgr.GetScheme(),
+			receiveAdapterImage: raImage,
+		},
+	}
+
+	return p.Add(mgr)
+}
+
+// reconciler reconciles a CloudEventIngress object
+type reconciler struct {
+	client              client.Client
+	scheme              *runtime.Scheme
+	receiveAdapterImage string
+}
+
+func (r *reconciler) InjectClient(c client.Client) error {
+	r.client = c
+	return nil
+}
+
+func (r *reconciler) Reconcile(ctx context.Context, object runtime.Object) error {
+	logger := logging.FromContext(ctx)
+
+	src, ok := object.(*v1alpha1.CloudEventIngress)
+	if !ok {
+		logger.Error("could not find CloudEventIngress source", zap.Any("object", object))
+		return nil
+	}
+
+	// This Source attempts to reconcile three things.
+	// 1. Determine the sink's URI.
+	// 2. Create a knative service
+	src.Status.InitializeConditions()
+
+	sinkURI, err := sinks.GetSinkURI(ctx, r.client, src.Spec.Sink, src.Namespace)
+	if err != nil {
+		src.Status.MarkNoSink("NotFound", "")
+		return err
+	}
+	src.Status.MarkSink(sinkURI)
+
+	_, err = r.getOwnedService(ctx, src)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			err = r.createKnativeService(ctx, src, sinkURI)
+			if err != nil {
+				logger.Error("Unable to create the receive adapter", zap.Error(err))
+				return err
+			}
+			src.Status.MarkDeployed()
+			return nil
+		}
+		return err
+	}
+
+	return nil
+}
+
+func (r *reconciler) createKnativeService(ctx context.Context, src *v1alpha1.CloudEventIngress, sinkURI string) error {
+	args := resources.CloudEventIngressArgs{
+		Image:   r.receiveAdapterImage,
+		Source:  src,
+		SinkURI: sinkURI,
+	}
+	expected := resources.MakeCloudEventIngress(&args)
+	logging.FromContext(ctx).Info("CloudEvent Object Created %s", expected)
+
+	if err := controllerutil.SetControllerReference(src, expected, r.scheme); err != nil {
+		return err
+	}
+	if err := r.client.Create(ctx, expected); err != nil {
+		return err
+	}
+	logging.FromContext(ctx).Info("CloudEventIngress created. %s", expected)
+	return nil
+}
+
+func (r *reconciler) getOwnedService(ctx context.Context, source *v1alpha1.CloudEventIngress) (*servingv1alpha1.Service, error) {
+	list := &servingv1alpha1.ServiceList{}
+	err := r.client.List(ctx, &client.ListOptions{
+		Namespace:     source.Namespace,
+		LabelSelector: labels.Everything(),
```

Comment:
Make a label selector with the labels you use when creating the Knative Service.